ignition-server/controllers/local_ignitionprovider.go

```diff
@@ -451,3 +472,62 @@ func copyFile(src, dst string) error {
 	}
 	return os.Chmod(dst, srcinfo.Mode())
 }
+
+// Finds the appropriate image reference in a multi-arch manifest image based on the current platform's OS and processor architecture
+func findImageRefByArch(ctx context.Context, imageRef string, pullSecret []byte, osToFind string, archToFind string) (manifestImageRef string, err error) {
+	manifestList, err := registryclient.GetManifest(ctx, imageRef, pullSecret)
+	if err != nil {
+		return "", fmt.Errorf("failed to retrieve manifest from image ref, %s: %w", imageRef, err)
+	}
+
+	_, payload, err := manifestList.Payload()
+	if err != nil {
+		return "", fmt.Errorf("failed to get manifest payload: %w", err)
+	}
+
+	m := new(manifestlist.DeserializedManifestList)
+	if err = m.UnmarshalJSON(payload); err != nil { 
+		return "", fmt.Errorf("failed to get unmarshalled manifest list: %w", err) 
+	}
+
+	matchingManifestForArch, err := findMatchingManifest(imageRef, m, osToFind, archToFind)
+	if err != nil {
+		return "", fmt.Errorf("failed to retrieve matching manifest for os/arch, %s/%s: %w", osToFind, archToFind, err)
+	}
+
+	return matchingManifestForArch, nil
+}
+
+func findMatchingManifest(imageRef string, deserializedManifestList *manifestlist.DeserializedManifestList, osToFind string, archToFind string) (string, error) {
+	matchingManifestForArch := imageRef
+	for _, manifestDesc := range deserializedManifestList.ManifestList.Manifests {
+		if osToFind == manifestDesc.Platform.OS && archToFind == manifestDesc.Platform.Architecture {
+			// Multi-arch image references look like either:
+			//	quay.io/openshift-release-dev/ocp-release@sha256:1a101ef5215da468cea8bd2eb47114e85b2b64a6b230d5882f845701f55d057f
+			//	quay.io/openshift-release-dev/ocp-release:4.11.0-0.nightly-multi-2022-07-12-131716
+			if strings.Contains(imageRef, "@sha") {
+				splitSHA := strings.Split(imageRef, "@")
+				if len(splitSHA) != 2 {
+					return "", fmt.Errorf("failed to parse imageRef %s", imageRef)
+				}
+
+				matchingManifestForArch = splitSHA[0] + "@" + string(manifestDesc.Descriptor.Digest)
+				log.Info("Found matching manifest of os/arch: " + matchingManifestForArch)
+				break
```

Comment:
please let's avoid if/else concatenation, this can be
```
			// Multi-arch image references look like either:
			//	quay.io/openshift-release-dev/ocp-release@sha256:1a101ef5215da468cea8bd2eb47114e85b2b64a6b230d5882f845701f55d057f
			//	quay.io/openshift-release-dev/ocp-release:4.11.0-0.nightly-multi-2022-07-12-131716
			if strings.Contains(imageRef, "@sha") {
				splitSHA := strings.Split(imageRef, "@")
				if len(splitSHA) != 2 {
					return "", fmt.Errorf("failed to parse imageRef %s", imageRef)
				}

				matchingManifestForArch = splitSHA[0] + "@" + string(manifestDesc.Descriptor.Digest)
				log.Info("Found matching manifest of os/arch: " + matchingManifestForArch)
				break
			}

			if strings.Contains(imageRef, "ocp-release:") {
				splitSHA := strings.Split(imageRef, ":")
				if len(splitSHA) != 2 {
					return "", fmt.Errorf("failed to parse imageRef %s", imageRef)
				}

				matchingManifestForArch = splitSHA[0] + "@" + string(manifestDesc.Descriptor.Digest)
				log.Info("Found matching manifest of os/arch: " + matchingManifestForArch)
				break
			}

			return "", fmt.Errorf("imageRef is an unknown format to parse, imageRef: %s", imageRef)
```

Reply:
This has been updated